client/my-sites/activity/filterbar/date-range-selector.jsx

```diff
@@ -288,51 +239,44 @@ export class DateRangeSelector extends Component {
 		const { isVisible } = this.props;
 		const from = this.getFromDate();
 		const to = this.getToDate();
+		const now = new Date();
 
 		const buttonClass = classnames( {
 			filterbar__selection: true,
 			'is-selected': !! from,
 			'is-active': isVisible && ! from,
 		} );
+
 		return (
-			<Fragment>
-				<Button
-					className={ buttonClass }
-					compact
-					borderless
-					onClick={ this.handleButtonClick }
-					ref={ this.dateRangeButton }
-				>
-					{ this.getFormattedDate( from, to ) }
-				</Button>
-				{ ( from || to ) && (
-					<Button
-						className="filterbar__selection-close"
-						compact
-						borderless
-						onClick={ this.handleResetSelection }
-					>
-						<Gridicon icon="cross-small" />
-					</Button>
-				) }
-				{ isWithinBreakpoint( '>660px' ) && (
-					<Popover
-						id="filterbar__date-range"
-						isVisible={ isVisible }
-						onClose={ this.handleClose }
-						position="bottom"
-						relativePosition={ { left: -125 } }
-						context={ this.dateRangeButton.current }
-					>
-						{ this.renderDateRangeSelection() }
-					</Popover>
-				) }
-				{ ! isWithinBreakpoint( '>660px' ) && (
-					<MobileSelectPortal isVisible={ isVisible }>
-						<Card>{ this.renderDateRangeSelection() }</Card>
-					</MobileSelectPortal>
+			<DateRangePicker
+				selectedStartDate={ from }
+				selectedEndDate={ to }
+				lastSelectableDate={ now }
+				onDateCommit={ this.handleDateRangeCommit }
+				renderTrigger={ props => (
+					<Fragment>
+						<Button
+							className={ buttonClass }
+							compact
+							borderless
+							onClick={ props.onTriggerClick }
+							ref={ props.buttonRef }
+						>
+							{ this.getFormattedDate( from, to ) }
+						</Button>
+						{ ( from || to ) && (
+							<Button
+								className="filterbar__selection-close"
+								compact
+								borderless
+								onClick={ this.handleResetSelection }
+							>
+								<Gridicon icon="cross-small" />
+							</Button>
```

Comment:
We should add some text here within a span for screenreaders. Should describe what happens when you click on the button.

Use https://github.com/Automattic/wp-calypso/tree/master/client/components/screen-reader-text

Reply:
Resolved in https://github.com/Automattic/wp-calypso/pull/30380